projects/social/social.py

```diff
@@ -57,8 +110,24 @@ def get_all_social_paths(self, user_id):
 
         The key is the friend's ID and the value is the path.
         """
-        visited = {}  # Note that this is a dictionary, not a set
-        # !!!! IMPLEMENT ME
+        
+        visited = {}
+        fof = []
+        visited[user_id] = [user_id]
+        for friend in self.friendships[user_id]:
```

Comment:
I believe you could condense this logic into one single while loop. Start at user_id -> check all neighbors individually -> has neighbor been added to visited? -> if no -> perform bfs on neighbor, push neighbor to a queue,  where you can later check that neighbors neighbors -> continue

Your current way of using bfs would work, but I would do that in a `for i in range(1, len(self.last_id) + 1)` find a path to each `i` from `user_id`. This will be less efficient, however, because you'll be checking islands that may not be connected at all to `user_id`